ProcessMaker/processmaker
# Resolve Orphaned Process Templates Generated During Editing in Modeler

## Issue & Reproduction Steps

The current behavior when editing a process template in the modeler results in the creation of new processes in the processes table instead of referencing the original process created during the first edit. This causes the accumulation of orphaned process templates, potentially leading to database clutter and performance degradation.

## Solution

To address this issue, the following steps will be taken:

1. A migration will be created to add the 'editing_process_uuid' column to the process_template table. This column will track the associated process when editing the template in the modeler.
2. During template editing in the modeler, a check will be performed to verify the presence of an 'editing_process_uuid' in the processes table. If the UUID exists, the process will be referenced. Otherwise, the template will be imported, creating a new process entry in the processes table.

## How to Test

1. Verify database access is available.
2. Navigate to the Templates tab and select a template for editing.
3. Confirm that the 'editing_process_uuid' is correctly set in the process_templates table.
4. Validate that the 'editing_process_uuid' in the process_templates table matches the UUID of the newly created process in the processes table.
5. Make an update to the template in the modeler and save the changes.
6. Edit the template again to trigger the process update.
7. Verify that no additional processes were created in the processes table and that the previous process UUID still matches the 'editing_process_uuid' in the process_templates table.


## Related Tickets & Packages
- [FOUR-8867](https://processmaker.atlassian.net/browse/FOUR-8867)

## Code Review Checklist


[FOUR-8867]: https://processmaker.atlassian.net/browse/FOUR-8867?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
